web/static/js/components/group_label_input.jsx

```diff
@@ -7,23 +7,63 @@ import styles from "./css_modules/group_label_input.css"
 // exceed the 1/4th column width on desktop, even with the widest possible characters
 const MAX_LENGTH_OF_GROUP_NAME = 20
 
-const GroupLabelInput = ({ groupWithAssociatedIdeasAndVotes, actions }) => (
-  <div className="ui fluid input">
-    <input
-      type="text"
-      className={styles.textInput}
-      placeholder="Add a group label"
-      defaultValue={groupWithAssociatedIdeasAndVotes.label}
-      maxLength={MAX_LENGTH_OF_GROUP_NAME}
-      onBlur={e => {
-        actions.submitGroupLabelChanges(groupWithAssociatedIdeasAndVotes, e.target.value)
-      }}
-    />
-    <div className="instruction">
-      <span className="keyboard-key">tab</span> to submit
-    </div>
-  </div>
-)
+class GroupLabelInput extends Component {
+  state = (() => {
```

Comment:
ahhh, given the destructuring you need to do here, it makes sense to house this in a es6 `constructor` function rather than in an immediately invoked function, as a) that'll pass in props as an argument, and b) it's more idiomatic of react class code when logic is necessary upon instantiation/construction. something like:

```es6
constructor(props) {
  const { groupWithAssociatedIdeasAndVotes: { label } } = props

  this.state = {
    oldLabel: label,
    showCheckmarkForLabelId: null,
  }
}
```

Reply:
Cool, didn't know any of that.  Modifying as per suggestion.